### PR TITLE
is_obstructed method one_sided

### PR DESCRIPTION
**Pull request by catherinezzy**
Hello team,

So the method only works if we are moving from left to right, down to up, or diagonally down to up. I could have done the other direction (see line 9) but then the code will be redundant. Please let me know if you have any ideas! 

Thank you!

### CODE REVIEW

File: app/models/piece.rb [hunk 1]
```diff
@@ -2,8 +2,58 @@ class Piece < ActiveRecord::Base
...
+
+
+	def occupied?(x, y)
+		game.pieces.where(x_cord: x, y_cord: y).present?
+	end	
+
+
+	def moving_direction(x, y)
+
+		if y_cord== y
```
Comment (AlekstheRealist) [inline app/models/piece.rb]: For the moving_direction method, what happens if y_cord == y and x_cord == x? Should we return an error value if this is ever the case?

File: app/models/piece.rb [hunk 2]
```diff
@@ -2,8 +2,58 @@ class Piece < ActiveRecord::Base
...
+			return 'horizontal'
+		elsif x_cord == x
+			return 'vertical'
+		else
+			return 'diagonal'
```
Comment (AlekstheRealist) [inline app/models/piece.rb]: What about cases where it's neither diagonal, vertical, nor horizontal like the example that Ilya gave during the chat? Maybe we can implement something like this into the if statement...
`if y_cord == y && x_cord == x`
  `return 'error'`
`elsif y_cord == y`
  `return 'horizontal'`
`elsif x_cord == x`
  `return 'vertical'`
`elsif #diagonal coordinate conditional statement`
  `return 'diagonal'`
`else`
  `return 'neither'`

File: app/models/piece.rb [hunk 3]
```diff
@@ -2,8 +2,58 @@ class Piece < ActiveRecord::Base
...
-	return false
-end
+	def is_obstructed?(x, y)
+
+		#check moving direction
+		path = moving_direction(x, y)
+
+		#path is horizontal from left to right
+		if path == 'horizontal' && x_cord < x
+			(x_cord + 1).upto(x - 1) do |changing_x|
```
Comment (AlekstheRealist) [inline app/models/piece.rb]: What happens if x_cord = 0 and x = 1 and it's implemented with the .upto callback? Will the .upto callback work with `(0 + 1).upto(1 - 1)`?

File: app/models/piece.rb (hunk 1 above)
Comment (catherinezzy) [inline app/models/piece.rb]: @AlekstheRealist Thank you for pointing that out. I will include this case scenario later.

File: app/models/piece.rb (hunk 3 above)
Comment (catherinezzy) [inline app/models/piece.rb]: @AlekstheRealist that is why I included "x_cord <x" on line 11. The other way around should be "x_cord>x"

File: app/models/piece.rb (hunk 2 above)
Comment (catherinezzy) [inline app/models/piece.rb]: good catch! I will redo my find direction method then.

File: app/models/piece.rb [hunk 4]
```diff
@@ -2,8 +2,58 @@ class Piece < ActiveRecord::Base
...
+				end
+			end
+		end
+
+		return false
+
+	end
+
+
+	def occupied?(x, y)
```
Comment (ilyakrasnov) [inline app/models/piece.rb]: Great abstraction!

**Comment by ilyakrasnov** [comment]
@catherinezzy There are some failing specs - please take a look

**Comment by ruishine** [comment]
Hi Catherine! Could you kindly fix tab formatting of this file to 2 space = 1 tab? Just for consistency with rest of code. I can fix when you merge, too!

**Comment by catherinezzy** [comment]
Hello Lin Rui,

I just changed it! I didn't know how that happened but it made my code really hard to read. Sorry!

**Comment by ilyakrasnov** [comment]
There are some failing tests - please have a look

**Comment by ilyakrasnov** [comment]
👍 lets merge this one!

**Pull request merged**